sources/lib/run-time/c-run-time.c

```diff
@@ -525,7 +525,7 @@ dylan_value primitive_raw_as_string (DBSTR buffer) {
   size_t len = strlen(buffer);
   size_t size = round_up_to_word(base_size + len + 1);
   dylan_value string = primitive_alloc_leaf_r(size, &KLbyte_stringGVKdW, len, 1);
-  memcpy(((dylan_byte_string*)string)->data, buffer, len);
+  memcpy(((dylan_byte_string*)string)->data, buffer, len + 1);
```

Comment:
What does the LLVM run-time do here?

Reply:
It copies the string, including the NUL.

Reply:
I'm pretty confident this is correct. This bug would only become apparent if you took in a string via the C interface and then used it again in a C function. I saw it in one of the tests where it gets the current shell via `getenv` and uses it to `exec` something.